include/dsn/internal/message_parser.h

```diff
@@ -137,10 +138,10 @@ namespace dsn
         bool _header_checked;
     public:
         dsn_message_parser(int buffer_block_size, bool is_write_only);
+        message_ex* receive_message_with_thrift_header(unsigned int read_length, /*out*/int& read_next);
```

Comment:
Thrift is not known to dsn message parser. better to have a new compound message parser instead.